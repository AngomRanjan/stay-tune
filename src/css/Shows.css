.shows__container {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  gap: 0.15em 0.1em;
  max-width: 960px;
  margin: 0 auto;
  padding: 0.25em;
}
