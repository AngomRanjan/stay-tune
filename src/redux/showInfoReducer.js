import axios from 'axios';

const GET_SHOW_INFO = 'GET_SHOW_INFO';

const initialState = [];

export const fetchShowInfo = (data) => ({
  type: GET_SHOW_INFO, payload: data,
});

export const getShowInfoAction = (BaseUrl) => async (dispatch) => {
  await axios.get(BaseUrl)
    .then((res) => {
      dispatch(fetchShowInfo(res.data));
    });
};

const showInfoReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_SHOW_INFO:
      return [action.payload];

    default:
      return state;
  }
};

export default showInfoReducer;
