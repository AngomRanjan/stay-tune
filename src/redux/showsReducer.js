import axios from 'axios';

const BaseUrl = 'https://api.tvmaze.com/shows';
const GET_SHOWS = 'GET_SHOWS';

const initialState = [];

export const fetchShows = (data) => ({
  type: GET_SHOWS, payload: data,
});

export const getShowsAction = () => async (dispatch) => {
  await axios.get(BaseUrl)
    .then((res) => {
      dispatch(fetchShows(res.data));
    });
};

const showsReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_SHOWS:
      return [...action.payload];
      // return [action.payload];

    default:
      return state;
  }
};

export default showsReducer;
