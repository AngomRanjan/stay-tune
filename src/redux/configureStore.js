import { configureStore, combineReducers } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import showsReducer from './showsReducer';
import showInfoReducer from './showInfoReducer';

const rootReducer = combineReducers({
  shows: showsReducer,
  showInfos: showInfoReducer.
});

const store = configureStore({
  reducer: rootReducer,
  middleware: [thunk],
});

export default store;
