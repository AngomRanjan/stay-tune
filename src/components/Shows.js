import '../css/Shows.css';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import Show from './Show';
import { getShowsAction } from '../redux/showsReducer';

const Shows = () => {
  const shows = useSelector((state) => state.shows);
  const dispatch = useDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(getShowsAction());
  }, []);

  return (
    <div
      className="shows"
    >
      <section className="shows__topBar">
        Space for top bar
      </section>
      <main className="shows__container">
        {shows.map((show) => (
          <Show
            key={show.id}
            id={show.id}
            name={show.name}
            image={show.image.medium}
            rating={show.rating.average}
            genres={show.genres[0]}
            onClick={() => navigate('/')}
          />
        ))}
      </main>
    </div>
  );
};

export default Shows;
