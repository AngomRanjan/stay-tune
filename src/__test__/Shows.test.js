import React from 'react';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import store from '../redux/configureStore';
import Shows from '../components/Shows';
import { fetchShows } from '../redux/showsReducer';

describe('Shows tests', () => {
  test('should render', () => {
    const detailpage = render(
      <Provider store={store}>
        <HashRouter>
          <Shows />
        </HashRouter>
      </Provider>,
    );
    expect(detailpage).toMatchSnapshot();
  });

  it('should return an action with type GET_SHOWS', () => {
    const data = [{
      id: 1,
      name: 'arfs',
    }];
    const action = fetchShows(data);
    expect(action.type).toBe('GET_SHOWS');
  });
});
