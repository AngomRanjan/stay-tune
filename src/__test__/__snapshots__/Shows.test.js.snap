// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="shows"
      >
        <section
          class="shows_banner banner"
          style="background-size: 100% 100%;"
        >
          <div
            class="banner__intro"
          >
            <article
              class="banner__intro__content"
            >
              <h1
                class="banner__intro__title"
              >
                Hi Welcome to Stay Tune!
              </h1>
              <p>
                Explore and enjoy from a list of collection.
              </p>
              <p>
                Start with this one....
              </p>
              <div
                class="showInfo__summary"
              >
                After a violent shipwreck, billionaire playboy Oliver Queen was missing and presumed dead for five years before being discovered alive on a remote island in the Pacific . He returned home to Starling City, welcomed by his devoted mother Moira , beloved sister Thea and former flame Laurel Lance. With the aid of his trusted chauffeur/bodyguard John Diggle , the computer-hacking skills of Felicity Smoak and the occasional , reluctant assistance of former police detective, now beat cop , Quentin Lance, Oliver has been waging a one-man war on crime.
              </div>
              <h4
                class="tagline"
              >
                What is next? ...
              </h4>
              <h5
                class="appName"
              >
                Stay Tune!
              </h5>
            </article>
          </div>
        </section>
        <section
          class="shows__search"
        >
          <form
            class="sheader__center"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <input
              id="search1"
              placeholder="Find your show"
              type="text"
              value=""
            />
          </form>
        </section>
        <main
          class="shows__container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="shows"
    >
      <section
        class="shows_banner banner"
        style="background-size: 100% 100%;"
      >
        <div
          class="banner__intro"
        >
          <article
            class="banner__intro__content"
          >
            <h1
              class="banner__intro__title"
            >
              Hi Welcome to Stay Tune!
            </h1>
            <p>
              Explore and enjoy from a list of collection.
            </p>
            <p>
              Start with this one....
            </p>
            <div
              class="showInfo__summary"
            >
              After a violent shipwreck, billionaire playboy Oliver Queen was missing and presumed dead for five years before being discovered alive on a remote island in the Pacific . He returned home to Starling City, welcomed by his devoted mother Moira , beloved sister Thea and former flame Laurel Lance. With the aid of his trusted chauffeur/bodyguard John Diggle , the computer-hacking skills of Felicity Smoak and the occasional , reluctant assistance of former police detective, now beat cop , Quentin Lance, Oliver has been waging a one-man war on crime.
            </div>
            <h4
              class="tagline"
            >
              What is next? ...
            </h4>
            <h5
              class="appName"
            >
              Stay Tune!
            </h5>
          </article>
        </div>
      </section>
      <section
        class="shows__search"
      >
        <form
          class="sheader__center"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <input
            id="search1"
            placeholder="Find your show"
            type="text"
            value=""
          />
        </form>
      </section>
      <main
        class="shows__container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
